docker/docker_image_dest.go

```diff
@@ -332,21 +332,31 @@ func (d *dockerImageDestination) TryReusingBlobWithOptions(ctx context.Context,
 	// Then try reusing blobs from other locations.
 	candidates := options.Cache.CandidateLocations2(d.ref.Transport(), bicTransportScope(d.ref), info.Digest, options.CanSubstitute)
 	for _, candidate := range candidates {
-		candidateRepo, err := parseBICLocationReference(candidate.Location)
-		if err != nil {
-			logrus.Debugf("Error parsing BlobInfoCache location reference: %s", err)
-			continue
+		var candidateRepo reference.Named
+		if !candidate.UnknownLocation {
+			candidateRepo, err = parseBICLocationReference(candidate.Location)
+			if err != nil {
+				logrus.Debugf("Error parsing BlobInfoCache location reference: %s", err)
+				continue
+			}
 		}
 		if candidate.CompressorName != blobinfocache.Uncompressed {
-			logrus.Debugf("Trying to reuse cached location %s compressed with %s in %s", candidate.Digest.String(), candidate.CompressorName, candidateRepo.Name())
+			logrus.Debugf("Trying to reuse cached location %s compressed with %s", candidate.Digest.String(), candidate.CompressorName)
 		} else {
-			logrus.Debugf("Trying to reuse cached location %s with no compression in %s", candidate.Digest.String(), candidateRepo.Name())
+			logrus.Debugf("Trying to reuse cached location %s with no compression", candidate.Digest.String())
 		}
 
 		// Sanity checks:
-		if reference.Domain(candidateRepo) != reference.Domain(d.ref.ref) {
-			logrus.Debugf("... Internal error: domain %s does not match destination %s", reference.Domain(candidateRepo), reference.Domain(d.ref.ref))
-			continue
+		if candidate.UnknownLocation || reference.Domain(candidateRepo) != reference.Domain(d.ref.ref) {
```

Comment:
- As it is right now, this is no longer just a sanity check
- The nesting of the `candidate.UnknownLocation vs. !candidate.UnknownLocation` is unnecessarily redundant.

If we accept a bit less elegant message text about the compression, how about something like:
```go
if !candidate.UnknownLocation {
   candidateRepo = parse
   sanity check for same domain
   logrus.Debugf("Trying to reuse cached location %{repo}@%{digest} (compression: %{candidate.CompressorName}")
} else 
   candidateRepo = …(d.ref.ref)
   logrus.Debugf("Trying to use cached digest %{digest} (no known location, compression: %{candidate.CompressorName}")
}